.gitattributes

```diff
@@ -0,0 +1,6 @@
+# Set all files to use line feed endings (since we can't match only ones without an extension)
+* eol=lf
+# And then reset all the files with extensions back to default
+*.* -eol
+
+*.sh text eol=lf
```

Comment:
This `.gitattributes` file "belongs" to another open PR, doesn't it? Better remove from this one?

Reply:
Yeah, I built on top of the .gitattributes PR so I could actually run the tests to time the improvement, and I expected that one to be an easy merge :|

Reply:
Meaning that... you'd wait for #1604 to be merged first?

Reply:
Well that was my plan (based on thinking it would "just work") but if you're ok with these changes specifically I can drop that commit and merge this by itself whilst I figure out how to make #1604 work properly on mac

Reply:
I don't want to rebase unnecessarily as it'll trigger unnecessary node-cli jobs

Reply:
I dropped the commit since that one has a bunch more to do it turns out